diff --git a/kernel/module/kmod.c b/kernel/module/kmod.c
index 0800d9891692..aeebc2550e3a 100644
--- a/kernel/module/kmod.c
+++ b/kernel/module/kmod.c
@@ -95,6 +95,8 @@ static int call_modprobe(char *orig_module_name, int wait)
 	argv[3] = module_name;	/* check free_modprobe_argv() */
 	argv[4] = NULL;
 
+	pr_info("%s: running %s %s %s %s\n", __func__, argv[0], argv[1], argv[2], argv[3]);
+
 	info = call_usermodehelper_setup(modprobe_path, argv, envp, GFP_KERNEL,
 					 NULL, free_modprobe_argv, NULL);
 	if (!info)
